spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskExecutionService.java

```diff
@@ -179,64 +191,82 @@ public long executeTask(String taskName, Map<String, String> taskDeploymentPrope
 
 		TaskLauncher taskLauncher = findTaskLauncher(platformName);
 
-		TaskDeployment existingTaskDeployment =
-				taskDeploymentRepository.findTopByTaskDefinitionNameOrderByCreatedOnAsc(taskName);
+		TaskDeployment existingTaskDeployment = taskDeploymentRepository
+				.findTopByTaskDefinitionNameOrderByCreatedOnAsc(taskName);
 		if (existingTaskDeployment != null) {
 			if (!existingTaskDeployment.getPlatformName().equals(platformName)) {
 				throw new IllegalStateException(String.format(
 						"Task definition [%s] has already been deployed on platform [%s].  " +
-						"Requested to deploy on platform [%s].",
+								"Requested to deploy on platform [%s].",
 						taskName, existingTaskDeployment.getPlatformName(), platformName));
 			}
 		}
+
 		TaskExecutionInformation taskExecutionInformation = taskExecutionInfoService
 				.findTaskExecutionInformation(taskName, taskDeploymentProperties, composedTaskRunnerName);
 
+		if (taskExecutionInformation.isComposed()) {
+			handleAccessToken(commandLineArgs, taskExecutionInformation);
+		}
+
 		TaskExecution taskExecution = taskExecutionRepositoryService.createTaskExecution(taskName);
 
-		if (taskExecutionInformation.isComposed()) {
-			boolean containsAccessToken = false;
-			boolean useUserAccessToken = false;
+		AppDeploymentRequest appDeploymentRequest = this.taskAppDeploymentRequestCreator.
+				createRequest(taskExecution, taskExecutionInformation, commandLineArgs, platformName);
 
-			final String dataflowServerAccessTokenKey = "dataflow-server-access-token";
-			final String dataflowServerUseUserAccessToken = "dataflow-server-use-user-access-token";
+		TaskManifest taskManifest = createTaskManifest(platformName, taskExecutionInformation, appDeploymentRequest);
 
-			for (String commandLineArg : commandLineArgs) {
-				if (commandLineArg.startsWith("--" + dataflowServerAccessTokenKey)) {
-					containsAccessToken = true;
-				}
-				if (StringUtils.trimAllWhitespace(commandLineArg).equalsIgnoreCase("--" + dataflowServerUseUserAccessToken + "=true")) {
-					useUserAccessToken = true;
-				}
-			}
+		TaskManifest previousManifest = this.dataflowTaskExecutionMetadataDao.getLatestManifest(taskName);
 
-			final String dataflowAccessTokenPropertyKey = "app." + taskExecutionInformation.getTaskDefinition().getRegisteredAppName() + "." + dataflowServerAccessTokenKey;
-			for (Map.Entry<String, String> taskDeploymentProperty : taskExecutionInformation.getTaskDeploymentProperties().entrySet()) {
-				if (taskDeploymentProperty.getKey().equals(dataflowAccessTokenPropertyKey)) {
-					containsAccessToken = true;
-				}
+		if(!taskDeploymentProperties.isEmpty()) {
+			taskDeploymentProperties = appDeploymentRequest.getDeploymentProperties();
```

Comment:
If I launch with `app.timestamp.format=yyyy`, `taskDeploymentProperties` passed to this function contains one entry. Deployment properties in an `appDeploymentRequest` is empty, thus `taskDeploymentProperties` is set to empty map. I think this is wrong and probably explains why relaunch don't get it.